process/transaction/shardProcess.go

```diff
@@ -311,56 +347,63 @@ func (txProc *txProcessor) processMoveBalance(
 		return err
 	}
 
-	err = txProc.checkIfValidTxToMetaChain(tx, acntSrc, adrDst)
-	if err != nil {
-		return err
-	}
-
 	txFee, err := txProc.processTxFee(tx, acntSrc, acntDst)
 	if err != nil {
 		return err
 	}
 
-	err = txProc.moveBalances(acntSrc, acntDst, tx.GetValue())
-	if err != nil {
-		return err
-	}
-
 	// is sender address in node shard
-	if acntSrc != nil {
+	if !check.IfNil(acntSrc) {
 		acntSrc.IncreaseNonce(1)
+		err := acntSrc.SubFromBalance(tx.Value)
+		if err != nil {
+			return err
+		}
+
+		err = txProc.accounts.SaveAccount(acntSrc)
+		if err != nil {
+			return err
+		}
 	}
 
-	txHash, err := core.CalculateHash(txProc.marshalizer, txProc.hasher, tx)
+	isPayable, err := txProc.scProcessor.IsPayable(adrDst)
```

Comment:
Could have checked `IsPayable` before subtracting from `acntSrc` perhaps?

Reply:
nope :). want to substract - and treat error of transactions after